index.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+if (process.env.NEW_RELIC_ENABLED) {
+  require('newrelic');
+}
+
+/* env Variables */
+var nodeEnv = process.env.NODE_ENV || 'development';
+var envDev = nodeEnv === 'development';
+var npmListKeyword = process.env.NPM_LIST_KEYWORD || 'yeoman-generator';
+var apiLimit = process.env.NPM_LIST_LIMIT || (envDev ? 100 : 5000);
+var updateInterval = process.env.UPDATE_INTERVAL_IN_SECONDS || 3610;
+var projectId = process.env.GCLOUD_PROJECT_ID || 'yo-gen-list';
+var bucket = process.env.GCLOUD_BUCKET || 'yeoman-generator-list';
+var keyFilename = process.env.GCLOUD_KEYFILENAME;
+var log = process.env.LOGGER || console;
+
+var fs = require('fs');
+var gcloud = require('gcloud');
+var q = require('q');
+var pluginCache = require('./plugin-cache')
+
+var gconfig = {
+  projectId: projectId
+};
+if (keyFilename) {
+  gconfig.keyFilename = keyFilename;
+}
+var bucket = gcloud(gconfig).storage().bucket(bucket)
+
+/* Plugin Cache operations */
+var Plugins = new pluginCache(npmListKeyword, apiLimit);
+var update = function () {
+  Plugins.update().finally(function () {
+    var file = bucket.file('cache.json');
+    Plugins.getCacheStream()
+    .pipe(file.createWriteStream({
+      gzip: true,
+      metadata: {
+        contentType: 'application/json',
+        cacheControl: 'max-age=3600',
```

Comment:
This is the default for public files, according to https://cloud.google.com/storage/docs/gsutil/commands/defacl

> By default, publicly readable objects are served with a Cache-Control header allowing such objects to be cached for 3600 seconds.